include/mbedtls/aes.h

```diff
@@ -69,21 +69,41 @@ extern "C" {
 // Regular implementation
 //
 
+#if defined(MBEDTLS_PADLOCK_C)
+/** VIA padlock needs 16 extra Bytes for alignment. */
+#define MBEDTLS_PADLOCK_EXTRA 4
+#else
+#define MBEDTLS_PADLOCK_EXTRA 0
+#endif
+
+#if defined(MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH)
+/** No extra Bytes needed for 128-bit key. */
+#define MBEDTLS_AES_RK_EXTRA (0 + MBEDTLS_PADLOCK_EXTRA)
+#else
+/** Allocate 80 extra Bytes for 192-bit and 256-bit key.
+ *   - 64 extra Bytes for key expansion
+ *   See https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.197.pdf
+ *      - Section 5, Nr = Nk + 6
+ *      - Section 5.2, the length of round keys is Nb*(Nr+1)
+ *   - 16 extra Bytes to simplify key expansion in the 256-bit case.
+ */
+#define MBEDTLS_AES_RK_EXTRA (20 + MBEDTLS_PADLOCK_EXTRA)
+#endif
+
 /**
  * \brief The AES context-type definition.
  */
 typedef struct mbedtls_aes_context {
     int MBEDTLS_PRIVATE(nr);                     /*!< The number of rounds. */
     size_t MBEDTLS_PRIVATE(rk_offset);           /*!< The offset in array elements to AES
                                                     round keys in the buffer. */
-    uint32_t MBEDTLS_PRIVATE(buf)[68];           /*!< Unaligned data buffer. This buffer can
-                                                    hold 32 extra Bytes, which can be used for
-                                                    one of the following purposes:
-                                                    <ul><li>Alignment if VIA padlock is
-                                                    used.</li>
-                                                    <li>Simplifying key expansion in the 256-bit
-                                                    case by generating an extra round key.
-                                                    </li></ul> */
+    /** Data buffer to hold round keys. This buffer can hold 16 / 80 / 96
+     *  extra Bytes, which can be used for one of the following purposes:
+     *   - Alignment if VIA padlock is used.
+     *   - Simplifying key expansion in the 256-bit case by generating
+     *   an extra round key.
+     */
+    uint32_t MBEDTLS_PRIVATE(buf)[44 + MBEDTLS_AES_RK_EXTRA];
```

Comment:
I'm not sure how necessary this PR is, given the amount of work we need to do, but if you want to do something like this I would expect to see
```suggestion
    uint32_t MBEDTLS_PRIVATE(buf)[MBEDTLS_AES_MAX_ROUNDS * MBEDTLS_AES_RK_UINT32S + MBEDTLS_AES_RK_EXTRA];
```

Reply:
I was considering the case when alignment is not needed but 256-bit keys used in AES. So this PR could still save 16 bytes without performance hit. But this PR still needs some work. Thanks for the suggestions.